mono/metadata/w32file-unix.c

```diff
@@ -2331,27 +2334,32 @@ write_file (gint src_fd, gint dest_fd, struct stat *st_src, gboolean report_erro
 	return TRUE ;
 }
 
-#if HAVE_COPYFILE_H
+#if HOST_DARWIN
 static int
-_wapi_copyfile(const char *from, const char *to, copyfile_state_t state, copyfile_flags_t flags)
+_wapi_clonefile(const char *from, const char *to, int flags)
 {
-	gchar *located_from, *located_to;
-	int ret;
+	if (clonefile_ptr != NULL) {
+		gchar *located_from, *located_to;
+		int ret;
 
-	located_from = mono_portability_find_file (from, FALSE);
-	located_to = mono_portability_find_file (to, FALSE);
+		located_from = mono_portability_find_file (from, FALSE);
+		located_to = mono_portability_find_file (to, FALSE);
 
-	MONO_ENTER_GC_SAFE;
-	ret = copyfile (
-		located_from == NULL ? from : located_from,
-		located_to == NULL ? to : located_to,
-		state, flags);
-	MONO_EXIT_GC_SAFE;
+		MONO_ENTER_GC_SAFE;
+		ret = clonefile_ptr (
+			located_from == NULL ? from : located_from,
```

Comment:
To avoid worrying about precedence I like to reverse this:
`foo ? foo : other`

But I realize this is how it was.

Reply:
I prefer to keep it unchanged to keep the diff smaller. I'll remember your advice when writing new code.